### PR TITLE
`r\mssql_virtual_machine`: add state refresh for `auto_patching`

### PR DESCRIPTION
**Pull request by myc2h6o**
Found from intermittent failure of `TestAccMsSqlVirtualMachine_autoPatching`  https://github.com/hashicorp/terraform-provider-azurerm/pull/15835#issuecomment-1109576028, apply same workaround as `auto_backup`

### CODE REVIEW

File: internal/services/mssql/mssql_virtual_machine_resource.go [hunk 1]
```diff
@@ -706,14 +769,14 @@ func flattenSqlVirtualMachineAutoPatching(autoPatching *sqlvirtualmachine.AutoPa
 		return []interface{}{}
 	}
 
-	var startHour int32
+	var startHour int
 	if autoPatching.MaintenanceWindowStartingHour != nil {
-		startHour = *autoPatching.MaintenanceWindowStartingHour
+		startHour = int(*autoPatching.MaintenanceWindowStartingHour)
```
Comment (myc2h6o) [inline internal/services/mssql/mssql_virtual_machine_resource.go:774]: This is for prop validation. the prop retrieved from d.Get() is and interface{} of int, which will be different from int32

**Comment by myc2h6o** [comment]
Successive test result
imageimage
image

**Review by katbyte — APPROVED** [review]
LGTM 🌻

**Comment by myc2h6o** [comment]
Hi @katbyte thanks for reviewing the pr! I've resolved the merge conflict, please take a look. Below is the updated test result

**Comment by myc2h6o** [comment]
Updated test result
image

**Pull request merged**